Examples/Scripts/PreFreeSurferPipelineBatchNHP.sh

```diff
@@ -0,0 +1,263 @@
+#!/bin/bash 
+
+
+Usage () {
+echo "$(basename $0) --StudyFolder=<path> --Subject=<id> --Species=<species> --RunMode=<mode> [--BrainExtract=<method>] [--StrucRes=<resolution>]"
+echo ""
+echo "Required Options:"
+echo "  --StudyFolder: Path to the study folder containing subject data"
+echo "  --Subject: Subject identifier"
+echo "  --Species: Species type (Human, Chimp, Mac, Marmoset, etc.)"
+echo "  --RunMode: Pipeline run mode (Default, ACPCAlignment, BrainExtraction, T2wToT1wRegAndBiasCorrection, AtlasRegistration)"
+echo ""
+echo "Optional Options:"
+echo "  --BrainExtract: INVIVO (default) or EXVIVO"
+echo "  --StrucRes: Structural resolution in mm (species-specific default will be used if not specified)"
+echo ""
+exit 1;
+}
+
+# Initialize variables
+StudyFolder=""
+Subjlist=""
+SPECIES=""
+RunMode=""
+BrainExtract=""
+StrucRes=""
+
+# Parse command line arguments
+get_batch_options() {
+    local arguments=("$@")
+    
+    local index=0
+    local numArgs=${#arguments[@]}
+    local argument
+    
+    while [ ${index} -lt ${numArgs} ]; do
+        argument=${arguments[index]}
+        
+        case ${argument} in
+            --StudyFolder=*)
+                StudyFolder=${argument#*=}
+                index=$(( index + 1 ))
+                ;;
+            --Subject=*)
+                Subjlist=${argument#*=}
+                index=$(( index + 1 ))
+                ;;
+            --Species=*)
+                SPECIES=${argument#*=}
+                index=$(( index + 1 ))
+                ;;
+            --RunMode=*)
+                RunMode=${argument#*=}
+                index=$(( index + 1 ))
+                ;;
+            --BrainExtract=*)
+                BrainExtract=${argument#*=}
+                index=$(( index + 1 ))
+                ;;
+            --StrucRes=*)
+                StrucRes=${argument#*=}
+                index=$(( index + 1 ))
+                ;;
+            *)
+                echo ""
+                echo "ERROR: Unrecognized Option: ${argument}"
+                echo ""
+                Usage
+                ;;
+        esac
+    done
+}
+
+# Parse arguments
+get_batch_options "$@"
+
+# Check required parameters
+if [ -z "$StudyFolder" ] || [ -z "$Subjlist" ] || [ -z "$SPECIES" ] || [ -z "$RunMode" ]; then
+    echo "ERROR: Missing required parameters"
+    Usage
+fi
+
+# Set default values for optional parameters
+BrainExtract=${BrainExtract:-INVIVO}
+RunMode=${RunMode:-Default}
+
+
+echo "$(basename $0) $@"
+
+if [ -z ${EnvironmentScript} ] ; then
+	EnvironmentScript="$HCPPIPEDIR/Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+source $EnvironmentScript
+
+# species specific config
+# If StrucRes is not provided, SetUpSPECIES.sh will use species-specific default
+source "$HCPPIPEDIR"/Examples/Scripts/SetUpSPECIES.sh "$SPECIES" "$StrucRes"
+RunMode=${RunMode:-Default}
+
+for Subject in $Subjlist ; do
+  echo $Subject
+
+  #Scan settings
+  source ${StudyFolder}/${Subject}/RawData/hcppipe_conf.txt
```

Comment:
We should probably add a comment explaining what variables this defines for us, so other users can either define them manually or set up similar files.

We could also consider making it conditional on whether the file exists and having lines to set them manually in this batch script.